mesonpy/_editable.py

```diff
@@ -0,0 +1,91 @@
+import functools
+import importlib.abc
+import os
+import shutil
+import subprocess
+import sys
+
+from types import ModuleType
+from typing import List, Optional, Union
+
+if sys.version_info >= (3, 9):
+    from collections.abc import Sequence
+else:
+    from typing import Sequence
+
+
+# This file should be standalone!
+# It is copied during the editable hook installation.
+
+
+class MesonpyFinder(importlib.abc.MetaPathFinder):
+    """Custom loader that whose purpose is to detect when the import system is
+    trying to load our modules, and trigger a rebuild. After triggering a
+    rebuild, we return None in find_spec, letting the normal finders pick up the
+    modules.
+    """
+
+    def __init__(
+        self,
+        project_path: str,
+        import_paths: List[str],
+        top_level_modules: List[str],
+        rebuild_commands: List[List[str]],
+    ) -> None:
+        if not os.path.isabs(project_path):
+            raise ImportError('Project path must be absolute')
+        self._project_path = project_path
+        self._import_paths = import_paths
+        self._top_level_modules = top_level_modules
+        self._rebuild_commands = rebuild_commands
+
+    def __repr__(self) -> str:
+        return f'{self.__class__.__name__}({self._project_path})'
+
+    @functools.lru_cache(maxsize=1)
+    def rebuild(self) -> None:
+        # delete old install directory
+        shutil.rmtree(self._project_path)
```

Comment:
Is there a reason we are deleting the install directory? 
It seem easier to re-use it and pass ``--only-changed`` to the install command?

Reply:
That makes sense.